pkg/scheduler/internal/cache/snapshot.go

```diff
@@ -193,6 +196,7 @@ func (s *Snapshot) Get(nodeName string) (*framework.NodeInfo, error) {
 	return nil, fmt.Errorf("nodeinfo not found for node name %q", nodeName)
 }
 
-func (s *Snapshot) IsPVCUsedByPods(key string) bool {
-	return s.usedPVCSet.Has(key)
+// PodsUsingPVC returns the set of scheduled pod names using the PVC.
+func (s *Snapshot) PodsUsingPVC(key string) sets.Set[string] {
```

Comment:
Is this enough?

```suggestion
func (s *Snapshot) PodsUsingPVC(key string) int {
```

Reply:
We need the pod names so we don't double-count pods if they have more than one conflicting volume with the pod-to-be-scheduled. Once we have the unique set of conflicting pod names (in `calPreFilterState`) then we can just grab the `.Len()` for `numConflictingPods`.